app/jobs/targets_cleanup_job.rb

```diff
@@ -0,0 +1,13 @@
+class TargetsCleanupJob < ApplicationJob
+  queue_as :default
+
+  after_perform do |job|
+    self.class.set(wait: 7.days).perform_later(job.arguments)
+  end
+
+  def perform
+    Target.find_each do |target|
+      target.destroy! if target.created_at + 7 < Time.zone.now
```

Comment:
The time condition may go inside the Target model. wdyt?

Reply:
Done!